### ISSUE TITLE
vagrant not set private_network or lost private_network after set

### ISSUE THREAD

**Issue by eromoe** [mitchellh/vagrant#7995]
It happen on centos 7 box.

Vagrantfile:
```
name = 'centos-dr'

Vagrant.configure("2") do |config|
    config.vm.define name do |machine|
        machine.vm.box = "centos7"
        machine.ssh.username = 'root'
        machine.ssh.password = 'vagrant'
        machine.ssh.insert_key = true
        machine.vm.hostname = name
        machine.vm.network :private_network, ip: "192.168.3.100"
        machine.vm.provider "virtualbox" do |v|
            v.name = name
        end
    end
end

```

1. Sometimes it doesn't set private_network

- I can't `ping` 192.168.3.100
- vagrant ssh, use ifconfig, no `192.168.3.100`
- not show in `ip route`, only two record :
```
[root@centos-dr ~]# ip route
default via 10.0.2.2 dev enp0s3  proto static  metric 100
10.0.2.0/24 dev enp0s3  proto kernel  scope link  src 10.0.2.15  metric 100
```

2. Set private_network successful, but lost it after a little time.

vagrant up
```
➜  vagrant up
Bringing machine 'centos-dr' up with 'virtualbox' provider...
==> centos-dr: Importing base box 'centos7'...
==> centos-dr: Matching MAC address for NAT networking...
==> centos-dr: Setting the name of the VM: centos-dr
==> centos-dr: Clearing any previously set forwarded ports...
==> centos-dr: Fixed port collision for 22 => 2222. Now on port 2206.
==> centos-dr: Clearing any previously set network interfaces...
==> centos-dr: Preparing network interfaces based on configuration...
    centos-dr: Adapter 1: nat
    centos-dr: Adapter 2: hostonly
==> centos-dr: Forwarding ports...
    centos-dr: 22 (guest) => 2206 (host) (adapter 1)
==> centos-dr: Booting VM...
==> centos-dr: Waiting for machine to boot. This may take a few minutes...
    centos-dr: SSH address: 127.0.0.1:2206
    centos-dr: SSH username: root
    centos-dr: SSH auth method: password
    centos-dr:
    centos-dr: Inserting generated public key within guest...
    centos-dr: Removing insecure key from the guest if it's present...
    centos-dr: Key inserted! Disconnecting and reconnecting using new SSH key...
==> centos-dr: Machine booted and ready!
==> centos-dr: Checking for guest additions in VM...
    centos-dr: The guest additions on this VM do not match the installed version of
    centos-dr: VirtualBox! In most cases this is fine, but in rare cases it can
    centos-dr: prevent things such as shared folders from working properly. If you see
    centos-dr: shared folder errors, please make sure the guest additions within the
    centos-dr: virtual machine match the version of VirtualBox you have installed on
    centos-dr: your host and reload your VM.
    centos-dr:
    centos-dr: Guest Additions Version: 4.3.28
    centos-dr: VirtualBox Version: 5.1
==> centos-dr: Setting hostname...
==> centos-dr: Configuring and enabling network interfaces...
==> centos-dr: Mounting shared folders...
    centos-dr: /vagrant => /root/centos7
```
vagrant ssh

```
➜  vagrant ssh
Welcome to your Vagrant-built virtual machine.
[root@centos-dr ~]# ifconfig
enp0s3: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.0.2.15  netmask 255.255.255.0  broadcast 10.0.2.255
        inet6 fe80::a00:27ff:fede:e0e  prefixlen 64  scopeid 0x20<link>
        ether 08:00:27:de:0e:0e  txqueuelen 1000  (Ethernet)
        RX packets 755  bytes 85056 (83.0 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 496  bytes 72366 (70.6 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

enp0s8: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.3.100  netmask 255.255.255.0  broadcast 192.168.3.255
        inet6 fe80::a00:27ff:fedb:3a66  prefixlen 64  scopeid 0x20<link>
        ether 08:00:27:db:3a:66  txqueuelen 1000  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 16  bytes 1980 (1.9 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 0  (Local Loopback)
        RX packets 4  bytes 328 (328.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 4  bytes 328 (328.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

[root@centos-dr ~]# exit
logout
Connection to 127.0.0.1 closed.
```
private_network was set correctly now.
But soon  lost: 
```
➜  ssh-copy-id centos-dr
The authenticity of host 'centos-dr (192.168.3.100)' can't be established.
ECDSA key fingerprint is 75:06:22:b7:e2:3b:46:88:51:97:c4:4f:27:20:21:26.
Are you sure you want to continue connecting (yes/no)? yes
```
Stucked at here ,  ctrul+c and vagrant ssh to see what happen:

```
➜  vagrant ssh
Last login: Fri Nov 11 17:17:15 2016 from 10.0.2.2
Welcome to your Vagrant-built virtual machine.
[root@centos-dr ~]# ifconfig
enp0s3: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.0.2.15  netmask 255.255.255.0  broadcast 10.0.2.255
        inet6 fe80::a00:27ff:fede:e0e  prefixlen 64  scopeid 0x20<link>
        ether 08:00:27:de:0e:0e  txqueuelen 1000  (Ethernet)
        RX packets 812  bytes 92235 (90.0 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 529  bytes 77823 (75.9 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

enp0s8: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet6 fe80::a00:27ff:fedb:3a66  prefixlen 64  scopeid 0x20<link>
        ether 08:00:27:db:3a:66  txqueuelen 1000  (Ethernet)
        RX packets 19  bytes 3839 (3.7 KiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 38  bytes 7129 (6.9 KiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 0  (Local Loopback)
        RX packets 4  bytes 328 (328.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 4  bytes 328 (328.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

[root@centos-dr ~]# ip route
default via 10.0.2.2 dev enp0s3  proto static  metric 100
10.0.2.0/24 dev enp0s3  proto kernel  scope link  src 10.0.2.15  metric 100
```
The already set private_network gone.

3. `vagrant reload` get that private_network back

**Comment by madduci** [comment]
I have another issue, always with a CentOS 7.2 box, using vagrant 1.8.7 and Virtualbox 5.1.8 on Arch Linux.

The setup works fine when I'm adding the boxes, but if I stop the boxes with vagrant halt and then I restart them with vagrant up, vagrant cannot connect to the virtual machine anymore. I have to destroy the boxes and load them again.

**Comment by igieon** [comment]
What happens from us is that we use centos 7.3 and what is problem with privet_network
configuration is:
 cat /etc/sysconfig/network-scripts/ifcfg-enp0s8
```
#VAGRANT-BEGIN
# The contents below are automatically generated by Vagrant. Do not modify.
NM_CONTROLLED=no
BOOTPROTO=none
ONBOOT=yes
IPADDR=192.168.56.137
NETMASK=255.255.255.0
DEVICE=enp0s8
PEERDNS=no
#VAGRANT-END
```

and from log vagrant wants to reload configuration via networkmanager but puts 
`NM_CONTROLLED=no` so centos networkManager ignores it.

**Comment by dharmab** [comment]
Experiencing this issue with a CentOS 7.3 box. Restarting the network service on the box works around the issue but is very inconvenient when working in a multi-machine project.

**Comment by igieon** [comment]
Just donwgrade to vagrant 1.9.0 where all is working (for me) or wait till is released 1.9.2 (where is change rollbacked)

**Comment by chrisroberts** [comment]
Hi there,

It looks like this has been resolved within a previously shipped version of Vagrant so I am now closing this issue. If the original issue was not fully resolved, please reopen this issue or create a new one.

Cheers!
